Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift

```diff
@@ -24,28 +26,36 @@ class SongAlgorithm{
                       let randomSong = items.randomElement(),
                       let randomURI = randomSong?["uri"]
                 else{
-                    NSLog("failed parse")
+                    NSLog("failed parsing random song dictionary response")
                     return completion(nil, error)
                 }
                 return completion(randomURI, error)
             }
         }
     }
     
+    func getRandomGenre() -> String? {
+        let genre = api_instance.genreSeedArray.randomElement()
+        return genre
+    }
+    
     //TODO: SPLIT INTO SMALLER HELPER FUNCTIONS FOR SAKE OF REUSE. CONSIDER MOVING PARTS OF THIS INTO API MANAGER
     func fetchSong(completion: @escaping ([String: Any]?, Error?) -> Void) {
-        let api_instance = SpotifyManager.shared()
-        //    I in the future will switch out this endpoint to access different search features
-        // currently just gets queries for 50 songs from a search of 'a'
-        let url = URL(string: "https://api.spotify.com/v1/search?q=a&type=track&limit=50")!
+        // currently queries 50 songs from a random genre, with a random offset
+        
+        let randomOffset = String(Int.random(in: 1..<800))
```

Comment:
Consider adding a seed to make this random num generator achieve more randomness. See this on how to add a seed: https://www.advancedswift.com/swift-random-numbers/ . I have often used the current timestamp as a seed since it is guaranteed you will get a different value when ever you call it